### PR TITLE
Fix/prompt studio basetool

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What

- Added workflow_filestorage to the Prompt Studio base tool to align with the SDK base tool property.
- Wrapped the feature under a feature flag (FeatureFlag.REMOTE_FILE_STORAGE) for controlled usage.
- The property may be refactored or removed later, based on its usage, once the feature flag is retired and the GA version of the SDK is adopted.

## Why

- The BaseTool class in the SDK includes this property, which is utilized in the x2text adapter.
- Ensuring the Prompt Studio base tool behaves consistently with the SDK BaseTool improves integration and maintainability.

## How

- Added the workflow_filestorage property to the Prompt Studio base tool.
- Integrated a check for the feature flag to conditionally initialize the property.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, under featureflag

## Database Migrations

- No

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/prompt_studio/prompt_studio_core_v2/prompt_ide_base_tool.py [hunk 1]
```diff
@@ -16,6 +19,13 @@ def __init__(self, log_level: LogLevel = LogLevel.INFO, org_id: str = "") -> Non
         """
         self.log_level = log_level
         self.org_id = org_id
+        self.workflow_filestorage = None
```
Comment (gaya3-zipstack) [inline backend/prompt_studio/prompt_studio_core_v2/prompt_ide_base_tool.py:22]: Hope this line will safe guard us from the exception
PromptIdeBaseTool' object has no attribute 'workflow_filestorage'

Comment (muhammad-ali-e) [reply backend/prompt_studio/prompt_studio_core_v2/prompt_ide_base_tool.py:22]: Yes, it will remain None until the feature flag is enabled.

**Review by harini-venkataraman — APPROVED** [review]
LGTM, added a minor comment

File: unstract/filesystem/src/unstract/filesystem/file_storage_config.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 def get_provider(var_name: str, default: str = "minio") -> FileStorageProvider:
     """Retrieve the file storage provider based on an environment variable."""
-    provider_name = os.environ.get(var_name, default).capitalize()
+    provider_name = os.environ.get(var_name, default).upper()
```
[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply unstract/filesystem/src/unstract/filesystem/file_storage_config.py:16]: It's enum key not value

**Review by vishnuszipstack — APPROVED** [review]

File: unstract/filesystem/src/unstract/filesystem/file_storage_config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gaya3-zipstack) [reply unstract/filesystem/src/unstract/filesystem/file_storage_config.py:16]: I am looking forward to using the common functions. Hoping to get that also in rc3...

**Review by gaya3-zipstack — APPROVED** [review]
Looks good.

**Pull request merged**